proposals/Exceptions.md

```diff
@@ -215,39 +216,57 @@ exception on top of the stack is popped and then thrown.
 
 ### Exception data extraction
 
-The `if_except` block begins with an `if_except` instruction, and has two
-instruction blocks,
-
-That is, the forms of an if_except block is:
+The `br_on_exn` instruction is a conditional branch that checks the exception
+tag of an exception on top of the stack, in the form of:
 
 ```
-if_except block_type except_index
-  Instruction*
-end
+br_on_exn label except_index
+```
+
+The `br_on_exn` instruction checks the exception tag of an `except_ref` on top
+of the stack if it matches the given exception index. If it does, it branches
+out to the label referenced by the instruction (In the binary form, the label
+will be converted to a relative depth immediate, like other branch
+instructions), and while doing that, pops exception values from the `except_ref`
```

Comment:
"pops exception values" -> "pops the `except_ref` value from the stack and instead pushes the exception's argument values"

Reply:
Done